catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/plugins/LocationUpdateField.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.video.stream.mpegts.plugins;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.alliance.libs.klv.GeometryOperator;
+import org.codice.alliance.libs.klv.GeometryUtility;
+
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.io.WKTReader;
+import com.vividsolutions.jts.io.WKTWriter;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.types.Core;
+
+/**
+ * Update the parent metacard location field with the union of each child location field.
+ */
+@NotThreadSafe
+public class LocationUpdateField implements UpdateParent.UpdateField {
+
+    private final GeometryOperator preUnionGeometryOperator;
+
+    private final GeometryOperator postUnionGeometryOperator;
+
+    private Geometry intermediateGeometry;
+
+    /**
+     *
+     * @param preUnionGeometryOperator applied to each child location before the union
+     * @param postUnionGeometryOperator applied to the location just before being saved to the parent
+     */
+    public LocationUpdateField(GeometryOperator preUnionGeometryOperator,
+            GeometryOperator postUnionGeometryOperator) {
+        this.preUnionGeometryOperator = preUnionGeometryOperator;
+        this.postUnionGeometryOperator = postUnionGeometryOperator;
+    }
+
+    @Override
+    public void updateField(Metacard parent, List<Metacard> children) {
+
+        WKTReader wktReader = new WKTReader();
```

Comment:
We've never injected in all the other places it's used. Do you want to make an argument as to why we should inject it? Also, WKTReader is a class, not an interface.